packages/mui-base/src/useAutocomplete/useAutocomplete.js

```diff
@@ -545,6 +545,17 @@ export function useAutocomplete(props) {
         isOptionEqualToValue(optionItem, valueItem),
       );
       if (itemIndex === -1) {
+        if (filterSelectedOptions && filteredOptions.length > 1) {
+          const nextItemToHighlightIndex = Math.min(
+            highlightedIndexRef.current,
+            filteredOptions.length - 1,
```

Comment:
`filteredOptions.length - 1` fallback might be a reason for  some of the failing testcases